src/S3/Parser/S3Parser.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+namespace Aws\S3\Parser;
+
+use Aws\Api\ErrorParser\XmlErrorParser;
+use Aws\Api\Parser\AbstractParser;
+use Aws\Api\Parser\Exception\ParserException;
+use Aws\Api\Service;
+use Aws\Api\StructureShape;
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Aws\ResultInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+
+class S3Parser extends AbstractParser
+{
+    /**
```

Comment:
for properties, I would just do inline docblocks like `/** @var XmlErrorParser */`